### ISSUE TITLE
Home Assistant Fehler "Die Entität hat keine Zustandsklasse mehr"

### ISSUE THREAD

**Issue by rt1970** [suaveolent/ha-hoymiles-wifi#81]
Die Entität hat keine Zustandsklasse mehr
Wir haben in der Vergangenheit Statistiken für „DTU AC-Tagesenergie“ (sensor.dtu_ac_tagesenergie) generiert, es verfügt jedoch nicht mehr über eine Zustandsklasse und daher können wir dafür keine langfristigen Statistiken mehr verfolgen.

Statistiken können erst erstellt werden, wenn diese Entität eine unterstützte Zustandsklasse hat.
Wenn die Zustandsklasse zuvor durch eine Integration bereitgestellt wurde, handelt es sich möglicherweise um einen Fehler. Bitte melde das Problem.
Solltest du die Zustandsklasse zuvor selbst festgelegt haben, korrigiere diese bitte. [Die verschiedenen Zustandsklassen und deren Verwendung findest du in der Entwicklerdokumentation.](https://developers.home-assistant.io/docs/core/entity/sensor/#long-term-statistics)
Wenn die Zustandsklasse dauerhaft entfernt wurde, möchtest du möglicherweise die entsprechenden Langzeitstatistiken aus deiner Datenbank löschen.
Möchtest du die Langzeitstatistiken von sensor.dtu_ac_tagesenergie dauerhaft aus deiner Datenbank löschen?

Ich will aber nichts löschen 🤔

**Comment by rt1970** [comment]
I have Hoymiles HMS-800W-2T.

Get-real-data-new Response:
device_serial_number: "414393xxxx"
timestamp: 1732541713
ap: 1
firmware_version: 1
sgs_data {
  serial_number: 220700xxxx
  firmware_version: 1
  voltage: 2344
  frequency: 4999
  active_power: 205
  current: 8
  power_factor: 999
  temperature: 178
  warning_number: 3
  crc_checksum: 28864
  link_status: 1
  modulation_index_signal: 9437290
}
pv_data {
  serial_number: 220700xxxx
  port_number: 1
  voltage: 316
  current: 34
  power: 108
  energy_total: 235010
  energy_daily: 171
  error_code: 50921472
}
pv_data {
  serial_number: 220700xxxx
  port_number: 2
  voltage: 318
  current: 34
  power: 108
  energy_total: 236861
  energy_daily: 172
  error_code: 50331648
}
dtu_power: 205
dtu_daily_energy: 343

**Comment by rt1970** [comment]
I have updatet HA to Version  2024.11.4. After that it will bring this Error Message.
What will HA delete? I dont understand this.

**Comment by suaveolent** [comment]
Can you post a screenshot of the DTU sensors? Do they still work?

**Comment by rt1970** [comment]
![HM](https://github.com/user-attachments/assets/a3f237ed-31e6-46d5-9a1d-89056baef345)
All work fine... But i dont have delete anything.

**Comment by suaveolent** [comment]
That’s weird. 
can you navigate to settings - entities and check if the above mentioned sensor appears twice?

**Comment by suaveolent** [comment]
Hm, that definitely seems like an issue with Home Assistant, since the integration does not provide any long term statistics. This is all automatically handled via HA.

**Comment by rt1970** [comment]
I dont know. But it message say:
We have generated statistics for DTU AC Daily Energy (sensor.dtu_ac_dailyenergy) in the past, but it no longer has a state class and therefore we can no longer track long-term statistics for it.
I show HA missing any attribute for daily energy!?

**Comment by rt1970** [comment]
> I will upgrade to latest HA and see if the error occurs here too.
Thank you!

total_increasing?
or
daily_increasing?

**Comment by suaveolent** [comment]
Total increasing. this is defined by HA. 

I noticed the latest release is 2024.11.3. did you upgrade to a pre-release?

**Comment by rt1970** [comment]
No. Only official.
Core   2024.11.3
Supervisor   2024.11.4
Operating System   13.2
Frontend   20241106.2

**Comment by suaveolent** [comment]
Right, I was just confused since you wrote 2024.11.4 (which of course is the supervisor).

After update to Core 2024.11.3 I don't see any error message for the mentioned entity. What happens if you restart HA? Is the error message still present? 

At the moment I am out of ideas, since I set the state correctly in code and I do not see the error message. (I can look back at data from may)
